spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWriteBuilder.java

```diff
@@ -130,8 +131,7 @@ public Write build() {
     Preconditions.checkArgument(handleTimestampWithoutZone || !SparkUtil.hasTimestampWithoutZone(table.schema()),
         SparkUtil.TIMESTAMP_WITHOUT_TIMEZONE_ERROR);
 
-    Schema writeSchema = SparkSchemaUtil.convert(table.schema(), dsSchema);
-    TypeUtil.validateWriteSchema(table.schema(), writeSchema, writeConf.checkNullability(), writeConf.checkOrdering());
```

Comment:
Updating Schema and commit  are not in transaction, this means that when writing data fails it will update schema, why not transaction？

Reply:
If a new column is seen, it doesn't matter if the write succeeds. That column will still need to be added to the table the next time the data is written.

Reply:
@rdblue Thanks for your reply, I agree you. Transaction is not important, If try to achieve  transaction during SparkWrite, there are some problems in sql, for example when I execute the sql"create table as select * from xxx" , it will throw "Cannot create a transaction within a transaction" question.  I think it is a disadvantage in sql to achieve transaction.